EIPS/eip-6384.md

```diff
@@ -0,0 +1,155 @@
+
+EIP-712 already formally binds an off-chain signature to a contract, with the "verifyingContract" parameter. We suggest adding a “view” function ("stateMutability":"view") to such contracts, that returns a human readable description of the meaning of this specific off-chain buffer.
+
+```
+/**
+ * @dev Returns the expected result of the offchain message.
+*/
+
+     function evalEIP712Buffer(bytes32 domainHash, string memory primaryType, bytes memory signature)
+     external
+     view
+     returns (string[] memory);{
+   ...
+
+}
+```
+
+**Every compliant contract MUST implement this function.**
+
+Using this function, wallets can submit the proposed off-chain signature to the contract and present the results to the user, allowing them to enjoy an “on-chain simulation equivalent” experience to their off-chain message.
+
+This function will have a well known name and signature, such that there is no need for updates in the EIP-712 structure.
+
+(Some suggested alternatives for the function name, such as "explainSignedMessage". We want to get the community feedback on the proper name)
+
+### Function's inputs
+
+The inputs of the function:
+
+- `domainHash` is the EIP-712's domainSeparator, a hashed `eip712Domain` struct.
+- `primaryType`is the EIP-712's `primaryType`.
+- `encodedSignature` is an ABI encoded message part of the EIP-712 full message.
+  We want to get the feedback of the community and specifically smart contract implementors on how they would like to receive this information: e.g. as seperate parameters as we show here or the fully ABI encoded message that includes all EIP-712 elements (including `types`).
+
+### Function's output(s)
+
+The output of the the function is an array of strings. The wallet SHOULD display them to its end-users. The wallet MAY choose to augment the returned strings with additional data. (e.g. resolve contract addresses to their name)
+
+The strings SHOULD NOT be formatted (e.g. should not contain HTML code) and wallets SHOULD treat this string as an untrusted input and handle its rendering as such.
+
+As it might be useful to provide a more structured output, we want to get the feedback of the community and specifically wallets' implementors on how they would like to receive it. One suggestion is that function would return structs that represent transfers and approvals to support a simulation like experience to the users. e.g. the `evalEIP712Buffer` function would return a strcut that represnts a transfer (contract, from, to, amount) and the wallet MAY display a resolved version of it to the end user. ("transfer 2 USDC to address 0x...")
+
+### Support for EIP-712 messages that are not meant to be used on-chain
+
+In case the EIP-712 message is not bounded to a contract, the wallet cannot call this function as the contract address is unkown and SHOULD default to its current display.
```

Comment:
You should define what "not bound to a contract" means here. I assume you mean `verifyingContract` in the domain separator?

```suggestion
If `verifyingContract` is not included in the EIP-712 domain separator, wallets MUST NOT retrieve a human-readable description using this EIP. In this case, wallets SHOULD fallback to their original EIP-712 display.
```